### ISSUE TITLE
Create documentation

### ISSUE THREAD

**Issue by jeff-j-chen** [jpconstantineau/BlueMicro_HID_Arduino_Library#1]
Hello, I greatly appreciate your project as it is allowing me to work with my Adafruit Feather perfectly. However, I think myself and other users would definitely appreciate documentation on the given functions. I tried my best to understand it from the given examples, but a readthedocs page with explanations for what KeyboardReport, Keypress, and all the other functions do and how to use them would be hihgly useful.

**Comment by jpconstantineau** [comment]
I agree that some docs would be very useful.  It's been started but sure need lots more work.  I would appreciate any help on that.

The functions maps relatively well with the ones in the Bluefruit library [here](https://github.com/adafruit/Adafruit_nRF52_Arduino/blob/master/libraries/Bluefruit52Lib/src/services/BLEHidAdafruit.cpp) as well as the TinyUSB library [here](https://github.com/adafruit/Adafruit_TinyUSB_Arduino/blob/master/src/arduino/hid/Adafruit_USBD_HID.cpp).  I don't know if they do have documentation somewhere for these but they would be a good starting point.

One of the key difference is that queues are being used between when the "send" functions are called and when the "processqueues" is used.  

I am currently refactoring my keyboard firmware code to use this library and will be able to put some more updates as I get the work progressing.

I would welcome any addition to the documentation or library.  Since this is an Arduino Library and that the whole repo gets included as part of the library, I would want to keep the documentation pretty lean (as in just using simple md files) as opposed to a bigger framework included in the repo to be published somewhere else.   I would welcome any suggestions how to proceed to keep it lean.

**Comment by jeff-j-chen** [comment]
You could create a simple [readthedocs](https://readthedocs.org/) page and make a link on the README, I find myself referencing them all the time, especially for Python libraries like [pyautogui](https://pyautogui.readthedocs.io/en/latest/). 

I don't know how much help I can be for writing documentation given that I'm having trouble understanding what the functions do and how to use them, and my knowledge of C++ is very limited. As I familiarize myself with this library I'll see if I can contribute.
